### PR TITLE
fix dualsol method and allow access to rhs and lhs of constraints

### LINKED ISSUE #78: properly handle dual values of bound constraints

### PR DESCRIPTION
**Pull request by mattmilten**
fix #78

### CODE REVIEW

**Comment by mattmilten** [comment]
This duplicates the behaviour of SCIP's internal implementation and is supposed to be removed with the next SCIP release #79

File: src/pyscipopt/scip.pyx [hunk 1]
```diff
@@ -1138,13 +1170,30 @@ cdef class Model:
...
+                PY_SCIP_CALL(SCIPgetConsVars(self._scip, transcons.cons, _vars, _nvars, &_success))
+                LPsol = SCIPvarGetLPSol(_vars[0])
+                rhs = SCIPgetRhsLinear(self._scip, transcons.cons)
+                lhs = SCIPgetLhsLinear(self._scip, transcons.cons)
+                if (LPsol == rhs) or (LPsol == lhs):
+                    dual = SCIPgetVarRedcost(self._scip, _vars[0])
+                    print(LPsol, rhs, lhs)
+                else:
+                    dual = 0
+
```
Comment (fserra) [inline src/pyscipopt/scip.pyx]: mmm There is a function `SCIPgetVarsLinear` that gives you the vars right away (so you don't need the malloc)

Also, should `SCIPgetVarRedcost` be 0 when `LPsol != rhs` and `LPsol != lhs` ?

The print should be removed

File: src/pyscipopt/scip.pyx [hunk 2]
```diff
@@ -1138,13 +1170,31 @@ cdef class Model:
...
                 dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
             else:
-                dual = SCIPgetDualsolLinear(self._scip, cons.cons)
+                _vars = SCIPgetVarsLinear(self._scip, transcons.cons)
+                LPsol = SCIPvarGetLPSol(_vars[0])
+                rhs = SCIPgetRhsLinear(self._scip, transcons.cons)
+                lhs = SCIPgetLhsLinear(self._scip, transcons.cons)
+                if (LPsol == rhs) or (LPsol == lhs):
+                    dual = SCIPgetVarRedcost(self._scip, _vars[0])
+
```
Comment (fserra) [inline src/pyscipopt/scip.pyx]: I still don't get why you have to check if the constraint is tight... if it is not tight the reduce cost should be 0 anyway..

**Comment by mattmilten** [comment]
Yeah, I guess you're right.

File: src/pyscipopt/scip.pyx (hunk 2 above)
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]: I just checked this and it actually isn't 0 for some reason. So we really need to do this extra check.

**Comment by mattmilten** [comment]
@fserra could you please accept the pull request?

**Comment by fserra** [comment]
I knew there was something i needed to do, but couldn't remember what it
was. thanks!

**Pull request merged**